### PR TITLE
Allow systematic exclusion of tracked identity

### LINKED ISSUE #1196: Turn on segment while in dev
## Tell us about the problem you're trying to solve
* We would like to emit segment events while in dev mode.
* We need a surefire way to not emit events if it is an airbyte person using the app.

### PR DESCRIPTION
**Pull request by michel-tricot**
## What
When tracking usage of the application accurately we need to exclude events generated by the team and some contributors. Right now we've been relying on people properly entering their emails but it is not reliable and we've been forgetting from time to time.

## How
Now we just need to define `AIRBYTE_ROLE` env variable in our shell rc and it will propagate to the application.
Tracking will still happen but there will be a new traits that we can exclude systematically

## Recommended reading order
1. `airbyte-analytics/src/main/java/io/airbyte/analytics/SegmentTrackingClient.java`
1. `docker-compose.yaml`
1. the rest

### CODE REVIEW

**Review by sherifnada — APPROVED** [review]

File: airbyte-analytics/src/main/java/io/airbyte/analytics/SegmentTrackingClient.java [hunk 1]
```diff
@@ -33,25 +33,30 @@
...
+                        final Supplier<String> roleSupplier) {
     this.identitySupplier = identitySupplier;
     this.analytics = analytics;
+    this.roleSupplier = roleSupplier;
   }
 
   public SegmentTrackingClient(Supplier<TrackingIdentity> identitySupplier) {
-    this.analytics = Analytics.builder(SEGMENT_WRITE_KEY).build();
-    this.identitySupplier = identitySupplier;
+    this(identitySupplier, Analytics.builder(SEGMENT_WRITE_KEY).build(), () -> System.getenv(AIRBYTE_ROLE.toUpperCase()));
```
Comment (cgardens) [inline airbyte-analytics/src/main/java/io/airbyte/analytics/SegmentTrackingClient.java:59]: Is there a reason you're not putting this `EnvConfigs`?

**Review by jrhizor — APPROVED** [review]
What value for `AIRBYTE_ROLE` should we put for our shells?

Also, are we still depending on completely turning it off in CI instead of setting a role?

**Review by cgardens — APPROVED** [review]
nice.

**Comment by michel-tricot** [comment]
@jrhizor yes, for MTU reasons. If we can use the same `customer_id` on every run that would be OK but right now we create a new UUID on every run.

I'll suggest a solution in a future PR.

File: airbyte-analytics/src/main/java/io/airbyte/analytics/SegmentTrackingClient.java (hunk 1 above)
Comment (michel-tricot) [reply airbyte-analytics/src/main/java/io/airbyte/analytics/SegmentTrackingClient.java]: fixed

**Pull request merged**